src/jsoncharutils.h

```diff
@@ -338,15 +337,15 @@ static inline uint64_t _umul128(uint64_t ab, uint64_t cd, uint64_t *hi) {
 
 really_inline value128 full_multiplication(uint64_t value1, uint64_t value2) {
   value128 answer;
-#ifdef SIMDJSON_REGULAR_VISUAL_STUDIO
+#if defined(SIMDJSON_REGULAR_VISUAL_STUDIO) || defined(SIMDJSON_IS_32BITS)
 #ifdef _M_ARM64
```

Comment:
if (32 bit) { if (ARM 64 bit) } is surprising to read! Is this because regular visual studio 64-bit with ARM needs a different version? This whole thing might read better without the nesting (#if / #elif / #else / #endif), but it's not a huge deal.

Reply:
I was trying to minimize the footprint of any changes, but yes, it's confusing.

This let's 32-bit use line 346 for `answer`. 

https://github.com/simdjson/simdjson/blob/88da62ba0930dbec09b0a41c8bcd02e88bf9f551/src/jsoncharutils.h#L338-L354

Reply:
Gotcha, minimized footprint. OK. So the _M_ARM64 is only there for VS.